examples/ShoppingAds/GetProductBiddingCategoryConstant.php

```diff
@@ -0,0 +1,186 @@
+    {
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        // Creates the query.
+        $query = 'SELECT product_bidding_category_constant.localized_name,
+            product_bidding_category_constant.product_bidding_category_constant_parent
+            FROM product_bidding_category_constant
+            WHERE product_bidding_category_constant.country_code IN (\'US\')
+            ORDER BY product_bidding_category_constant.localized_name ASC';
+
+        // Creates a map of top level categories.
+        $rootCategories = [];
+        // Creates a map of all categories found in the results.
+        // This is a convenience lookup to enable fast retrieval of existing categories.
+        $biddingCategories = [];
+
+        // Performs the search request.
+        $response = $googleAdsServiceClient->search(
+            $customerId,
+            $query,
+            ['pageSize' => self::PAGE_SIZE]
+        );
+
+        // Iterates over all rows in all pages to extract the result.
+        foreach ($response->iterateAllElements() as $googleAdsRow) {
+            /**
+             * @var GoogleAdsRow $googleAdsRow
+             * @var ProductBiddingCategoryConstant $productBiddingCategory
+             */
+            $productBiddingCategory = $googleAdsRow->getProductBiddingCategoryConstant();
+            $resourceName = $productBiddingCategory->getResourceName();
+
+            // Adds the category in the map if new.
+            if (!array_key_exists($resourceName, $biddingCategories)) {
+                $biddingCategories[$resourceName] = [];
+            }
+            // Ensures that the localized name attribute for the categories is set.
+            // It will not be initialized for new categories or categories added
+            // to biddingCategories as a result of being a parent category below.
+            $biddingCategories[$resourceName]['localizedName'] =
+                $productBiddingCategory->getLocalizedNameUnwrapped();
+
+            if ($productBiddingCategory->getProductBiddingCategoryConstantParent() != null) {
+                // If the category has a parent.
+                $parentResourceName =
+                    $productBiddingCategory->getProductBiddingCategoryConstantParentUnwrapped();
+                // Adds the parent category in the map if new.
+                if (!array_key_exists($parentResourceName, $biddingCategories)) {
+                    $biddingCategories[$parentResourceName] = [];
+                }
+                // Adds the category as a child category of the parent category.
+                $biddingCategories[$parentResourceName]['children'][$resourceName] =
+                    &$biddingCategories[$resourceName];
+            } else {
+                // Otherwise adds the category as a root category.
+                $rootCategories[$resourceName] = &$biddingCategories[$resourceName];
```

Comment:
Is the reference access (&) here needed? I've done some simply mapping in other examples before. It didn't seem we need it.

Reply:
I had to add them both to make it work as intended (one line printed per product bidding category constant). If I do not use references, only corner cases make it through (~500/7000).